crates/jstz_cli/src/sandbox/daemon.rs

```diff
@@ -457,3 +467,99 @@ pub async fn main(cfg: &mut Config) -> Result<()> {
     cfg.save()?;
     Ok(())
 }
+
+pub async fn main(no_daemon: bool, cfg: &mut Config) -> Result<()> {
+    if no_daemon {
+        run_sandbox(cfg).await?;
+    } else {
+        // Print banner
+        info!("{}", style(SANDBOX_BANNER).bold());
+        info!(
+            "        {} {}",
+            env!("CARGO_PKG_VERSION"),
+            styles::url(env!("CARGO_PKG_REPOSITORY"))
+        );
+        info!("");
+
+        let path = sandbox_daemon_log_path()?;
+        let stdout_file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            .truncate(true)
+            .open(path.clone())?;
+        let mut child = Command::new(std::env::current_exe()?)
+            .args(["sandbox", "start", "--no-daemon"])
+            .stdout(Stdio::from(stdout_file))
+            .spawn()?;
+
+        let file = File::open(&path)?;
+        let mut reader = BufReader::new(file);
+        let mut buffer = String::new();
+
+        let regex = Regex::new(r"\((\d+)\)").unwrap();
+
+        let mut progress: u32 = 0;
+
+        let progress_bar = ProgressBar::new(MAX_PROGRESS as u64);
+        progress_bar.set_style(
+            ProgressStyle::default_bar().template(
+                "{spinner:.green} [{bar:40.cyan/blue}] {pos:>7}/{len:7} {msg}",
+            )?,
+        );
+
+        loop {
+            reader.stream_position()?;
+
+            while reader.read_line(&mut buffer)? > 0 {
+                if let Some(captures) = regex.captures(&buffer) {
+                    if let Some(matched) = captures.get(1) {
+                        if let Ok(num) = matched.as_str().parse::<u32>() {
+                            progress = num;
+                            progress_bar.set_position(progress.into());
+                        }
+                    }
+                }
+                buffer.clear();
+            }
+
+            if progress == MAX_PROGRESS {
+                progress_bar.finish_with_message("Sandbox started 🎉");
+                break;
+            }
+
+            if let Ok(Some(status)) = child.try_wait() {
+                progress_bar
+                    .finish_with_message(format!("Sandbox failed to start: {:}", status));
+                break;
+            }
+
+            thread::sleep(Duration::from_millis(100));
```

Comment:
Note to self: the mix of tokio + threading is becoming dangerous (and causing issues)